zipline/sources/benchmark_source.py

```diff
@@ -35,16 +35,38 @@ def __init__(self, benchmark_sid, env, trading_days, data_portal,
 
             self._validate_benchmark()
 
-        self.precalculated_series = \
-            self._initialize_precalculated_series(
-                self.benchmark_sid,
-                self.env,
-                self.trading_days,
-                self.data_portal
-            )
+            self._precalculated_series = \
+                self._initialize_precalculated_series(
+                    self.benchmark_asset,
+                    self.env,
+                    self.trading_days,
+                    self.data_portal
+                )
+        else:
```

Comment:
we were passing an `int` into `data_portal.get_history_window`, and I wanted to avoid that.  Also cleaned up a bit of the initialization logic.